### PR TITLE
add TPCH and compileWorkUnit UT

### PR DESCRIPTION
**Pull request by jikunshang**

### CODE REVIEW

**Comment by jikunshang** [comment]
@yyang52 @winningsix Please take a review, thanks.

File: Cider/CiderExecutionKernel.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include <memory>
+#include <vector>
+
+#include <QueryEngine/InputMetadata.h>
+#include <QueryEngine/RelAlgExecutionUnit.h>
+
+class CiderExecutionKernel {
+ public:
+  virtual ~CiderExecutionKernel(){};
+  void runWithDataMultiFrag(const int8_t*** multi_col_buffers,
```
Comment (winningsix) [inline Cider/CiderExecutionKernel.h:28]: make it private?

File: Cider/CiderExecutionKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+#include "Shared/StringTransform.h"
+#include "Shared/scope.h"
+#include "SqliteConnector/SqliteConnector.h"
+
+class CiderExecutionKernelImpl;
+
+inline CiderExecutionKernelImpl* getImpl(CiderExecutionKernel* ptr) {
+  return (CiderExecutionKernelImpl*)ptr;
+}
+inline const CiderExecutionKernelImpl* getImpl(const CiderExecutionKernel* ptr) {
```
Comment (winningsix) [inline Cider/CiderExecutionKernel.cpp:38]: line break.

File: Cider/CiderExecutionKernel.cpp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+#include "QueryEngine/ExpressionRange.h"
+#include "QueryEngine/RelAlgExecutionUnit.h"
+#include "QueryEngine/ResultSetReductionJIT.h"
+#include "QueryRunner/QueryRunner.h"
+#include "Shared/DateConverters.h"
+#include "Shared/StringTransform.h"
+#include "Shared/scope.h"
+#include "SqliteConnector/SqliteConnector.h"
+
+class CiderExecutionKernelImpl;
```
Comment (winningsix) [inline Cider/CiderExecutionKernel.cpp:33]: it seems format needed for this file?

File: QueryEngine/ColumnIR.cpp [hunk 4]
```diff
@@ -111,8 +111,9 @@ std::vector<llvm::Value*> CodeGenerator::codegenColVar(const Analyzer::ColumnVar
   const int rte_idx = adjusted_range_table_index(col_var);
   CHECK_LT(static_cast<size_t>(rte_idx), cgen_state_->frag_offsets_.size());
   const auto catalog = executor()->getCatalog();
-  CHECK(catalog);
-  if (col_var->get_table_id() > 0) {
+//  CHECK(catalog);
+  // this if block is checking some geo related column info, skip it.
+  if (false) {
```
Comment (winningsix) [inline QueryEngine/ColumnIR.cpp:116]: Better to track previous code in comments like what you do for QueryMemoryDescriptor.cpp?

**Review by winningsix — APPROVED** [review]
LGTM. Some minor comments.

File: Cider/CiderExecutionKernel.cpp (hunk 3 above)
Comment (jikunshang) [reply Cider/CiderExecutionKernel.cpp:33]: these files are formatted, column limit is 90.

File: Cider/CiderExecutionKernel.h (hunk 1 above)
Comment (jikunshang) [reply Cider/CiderExecutionKernel.h:28]: I think this could be used.

**Pull request merged**